src/main/java/org/jabref/model/entry/identifier/DOI.java

```diff
@@ -108,12 +108,12 @@ public DOI(String doi) {
         Matcher matcher = EXACT_DOI_PATT.matcher(trimmedDoi);
         if (matcher.find()) {
             // match only group \1
-            this.doi = matcher.group(1);
+            this.doi = matcher.group(1).replaceAll("[^\\w,/,:,-,.,-]|[_]", "");
```

Comment:
```suggestion
            this.doi = matcher.group(1).replaceAll("[^\\w/:-.-_]", "");
```

Variable name: `ILLEGAL_DOI_CHARACTERS` --> `Pattern.compile`?

(devcall)